### ISSUE TITLE
WCDB.swift link error:  Undefined symbol: _$s9WCDBSwift10ColumnTypeO9integer32yA2CmFWC

### ISSUE THREAD

**Issue by pcjbird** [Tencent/wcdb#965]
### The language of WCDB

> e.g. Objective-C, Swift or Java 

Swift

### The version of WCDB

> e.g. v1.0.5

WCDB.swift 2.0.2.5

### The platform of WCDB

> e.g. iOS, macOS or Android

iOS

### The installation of WCDB

> e.g. Cocoapods, Carthage, Maven, AAR Package or Git clone

Cocoapods

### What's the issue?

> Post the outputs or screenshots for errors.
> 
> Explain what you want by example or code **in English**.

<img width="1552" alt="image" src="https://github.com/Tencent/wcdb/assets/6175452/0ccb67e0-44bb-4e64-9a25-7d566de1d05b">

**Comment by pcjbird** [comment]
when 

```swift
final class TestDBTable: TableCodable {
```

defined in a framework or library cause this link error, and defined in the final app works ok.

**Comment by Qiuwen-chen** [comment]
May be you should link WCDB to that framework or library.

**Comment by pcjbird** [comment]
> May be you should link WCDB to that framework or library.

the framework build success. but when invoked in the final app, the error occured, and if I remove table defines columntype defines, the link error is gone.  but I must define the table in my framework.

**Comment by Qiuwen-chen** [comment]
> > May be you should link WCDB to that framework or library.
> 
> the framework build success. but when invoked in the final app, the error occured, and if I remove table defines columntype defines, the link error is gone. but I must define the table in my framework.

This is a configuration problem of your project, not related to WCDB.

**Comment by pcjbird** [comment]
I tried my best to understand what happened. When I directly referenced WCDB.swift in the target program with CocoaPods, the compiled WCDBSwift.framework indeed did not include symbol information such as WCDBSwift10ColumnTypeO9integer32yA2CmFWC, which is defined in the SyntaxConst.swift file. However, this does not seem to affect its usage in the target program. I can only speculate that the compiler has optimized something, but I am not sure how exactly it was done.

In addition, I also tried to independently compile WCDBSwift.framework and used CocoaPods to compile it in another framework B project. After that, I found that the generated framework did include symbol information such as WCDBSwift10ColumnTypeO9integer32yA2CmFWC. 

![Longshot-20230727104312](https://github.com/Tencent/wcdb/assets/6175452/b734680a-99a7-4496-8b06-ada7d9fc7ffc)


This explains why when I referenced the compiled framework B in the target project and also used CocoaPods to reference WCDB.swift, it reported an "Undefined Symbol" error.

Can anyone help me understand what happened here?
